### ISSUE TITLE
Infinite file size ballooning, memory leaks, and crashes being caused by langflow.db

### ISSUE THREAD

**Issue by EDLLT** [langflow-ai/langflow#4008]
Related separate issue: https://github.com/langflow-ai/langflow/issues/3970

### Bug Description

- Deleting the entire flow clears up the relevant parts of the vertex_build cache; however, deleting the file component does not delete its relevant data from the vertex_build cache. This causes ``langflow.db`` to pile up space from non-existing components overtime within the flow

- When rebuilding the file component explicitly(doesn't matter if it's frozen or not), the previously cached data does not get removed from ``vertex_build`` in ``langflow.db`` and the data gets duplicated within vertex_build.

- Uploading the same file multiple times using langflow's file component increases memory usage. In my case, uploading the ``random_ascii_70MB.txt`` 10 times increased memory usage by \~1GB

- [Create a file component, upload 70MB file, use split text on it with default options and build the component **at least twice**(freezing/unfreezing makes no difference)](https://github.com/langflow-ai/langflow/issues/4008#issuecomment-2407550796)
    - Upon reloading the page, langflow spikes up in memory then [crashes the langflow frontend](https://github.com/user-attachments/assets/d605c586-39cd-411b-8ac9-f0c67210c5d9)

### Reproduction

Before installation, I created a python venv using
```bash
python3 -m venv langflow
source langflow/bin/activate
```
Then I installed langflow while in that environment
```
pip install langflow -U
```

Steps to reproduce

Please see both of these links
https://github.com/langflow-ai/langflow/issues/4008#issuecomment-2399700910
https://github.com/langflow-ai/langflow/issues/4008#issuecomment-2407550796

### Operating System

OS: Ubuntu 24.04.1 LTS x86_64

### Langflow Version

Langflow v1.0.18

### Python Version

3.12

### Screenshot

![image](https://github.com/user-attachments/assets/f907ed88-8bcd-4ffc-ac55-d25e6ce0cff5)

**Comment by EDLLT** [comment]
> Related: #3970
> 
> ### Bug Description
> Whenever I try to embed my 70MB text file, it overtime starts eating up all my RAM. Then even booting langflow begins eating up ram, I am assuming because of that 70MB file
> 
> Initially, I was attempting to perform RAG and noticed my system lagging heavily then I narrowed it down to the file component
> 
> ### Reproduction
> Before installation, I created a python venv using
> 
> ```shell
> python3 -m venv langflow
> source langflow/bin/activate
> ```
> 
> Then I installed langflow while in that environment
> 
> ```
> pip install langflow -U
> ```
> 
> Steps to reproduce
> 
> Command used to generate the file
> 
> ```shell
> head -c 70000000 /dev/zero | tr '\0' 'a' > random_ascii_70MB.txt
> ```
> 
> Create a normal "file" component in Langflow and select the file we had generated
> 
> Build the component then wait for it to build successfully then rebuild Notice that the total memory consumption of langflow increases by ~1GB after every successful build
> 
> Also, if I now go back to "My Collection" then attempt to open my flow, it spikes up my ram before I get a crash from langflow(image attached below).
> 
> ### Expected behavior
> Upon building the same "file" component over and over, the total memory usage of langflow shouldn't increase.
> 
> ### Who can help?
> _No response_
> 
> ### Operating System
> OS: Ubuntu 24.04.1 LTS x86_64
> 
> ### Langflow Version
> Langflow v1.0.18
> 
> ### Python Version
> 3.12
> 
> ### Screenshot
> ![image](https://private-user-images.githubusercontent.com/47005435/373332578-f907ed88-8bcd-4ffc-ac55-d25e6ce0cff5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgyODI5NzEsIm5iZiI6MTcyODI4MjY3MSwicGF0aCI6Ii80NzAwNTQzNS8zNzMzMzI1NzgtZjkwN2VkODgtOGJjZC00ZmZjLWFjNTUtZDI1ZTZjZTBjZmY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA3VDA2MzExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyYmVmOTE5N2U4Y2Q3NTUxZWU5NTg5NmVhZmYyZjk5ZTM5OWFkMmU2YmRkNjJiYjY3ZmMxZWUyNzZkOTRjZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LDM2szCSiQ_vyHfot9DYMQZ1U8tcprrrGpLsoYbUFsA)
> 
> ### Flow File
> _No response_

Through further experimentation, I discovered that even uploading is causing a memory leak.
If you simply upload the same file multiple times, you'll notice the memory usage increasing

**Comment by EDLLT** [comment]
I've tested running this multiple times while the other parts are commented out
```
data = parse_text_file_to_data(resolved_path, silent_errors)
```

It wasn't causing any memory leaks(after loading it, memory returned back to the exact same spot)

However, this was problematic
```
self.status = data if data else "No data"
```
It seems that this line is causing a memory leak as every time I rebuild the component with that line, memory doesn't seem to get back but rather increases on every build

**Comment by EDLLT** [comment]
Figured out something very problematic. The ``langflow.db`` file is storing all every component's output, it seems

![image](https://github.com/user-attachments/assets/a7434db9-9f39-4a75-a3f1-d10bda8217bb)

**Comment by EDLLT** [comment]
Seems like this file plays a huge role

``langflow_source-code/src/backend/base/langflow/services/database/models/vertex_builds/crud.py``
It commits to the db the output results of the component, I am assuming for caching upon rebuilds.
The problem is, it doesn't really cache properly as it caches the same file's content over and over again then committing to the db making it balloon

**Comment by EDLLT** [comment]
> Found more. What's the purpose of logging every vertex build? As it seems to be accumulating and taking up tremendous amounts of storage space in the db file causing langflow to crash

@nicoloboschi

**Comment by codenprogressive** [comment]
Hey @EDLLT, thanks for providing this step-by-step to reproduce the issue! 
I observed the same issue when processing huge volume of data. Looks like in both `vertex_build` table and `transaction` table both are logging the input and output of each component.
so if you are logging 70MB worth of text in one column, this will make the consumption of that table impossible. then if you delete those tables, everything will resume working, but you loose the metadata stored in langflow DB.
Hey @Cristhianzl can you please help with this issue?

**Comment by Cristhianzl** [comment]
hey @codenprogressive, @EDLLT how are you?

We have a fix for this error coming soon in the next release. The front end can now handle this amount of data without breaking, as it did for you, @EDLLT.

The point is that we can't truncate the data to save it in the database because it would break other features like the freeze/freeze path. We need to save the runs on `vertex_build` and `transaction` tables for these features to work properly.

So my advice is: when you're working with large files or large amounts of data, please try to use these features as well. This way, the file or large data won't be processed twice. :)

**Comment by Cristhianzl** [comment]
I'm closing this issue because the error has been fixed on the MAIN branch and will be included in the next Langflow release!

Thank you!!

**Comment by EDLLT** [comment]
@Cristhianzl Hey!
Thanks for the prompt fix.

I have tested it out and unfortunately the issues that I had mentioned in previous comments still occur and new issues that arise from the frontend

In these, I talk about the file component but I think the same issue occurs to every component that deals with data

### Issues that still occur
- Deleting the entire flow clears up the relevant parts of the vertex_build cache; however, deleting the file component does not delete its relevant data from the vertex_build cache. This causes ``langflow.db`` to pile up space from non-existing components overtime within the flow

- When rebuilding the file component explicitly(doesn't matter if it's frozen or not), the previously cached does not get removed from ``vertex_build`` in ``langflow.db`` and the data gets duplicated within vertex_build.

- Uploading the same file multiple times using langflow's file component increases memory usage. In my case, uploading it 10 times increased memory usage by ~1GB


### New issues
- When processing a large amount of data using the file component
    -  After it has built successfully, we are able to press on Data within langflow to view the contents of our file.
    - Then, when I try to refresh the page and try to view the data within my already built file component, it crashes my tab(tested on both brave and chrome) 
    - ![image](https://github.com/user-attachments/assets/8260edd1-e98a-40c7-a3f1-b90e0b04df0e)
    - ![image](https://github.com/user-attachments/assets/d605c586-39cd-411b-8ac9-f0c67210c5d9)

- Taking data from the file component then using split text processes successfully but
    - Upon reloading the page(on the flow not the main menu), langflow starts taking up a significant amount of ram(reached up to ~15GB) before it settles down and afterwards, the browser's memory starts spiking up to 2~3GB before crashing the tab(tested on both brave and chrome)

**Comment by Cristhianzl** [comment]
@EDLLT hi,

Are you using the freeze/freeze path feature to prevent reprocessing files that have already been processed? 

Are you on the main branch locally? Please note that we haven't released the fix for this yet.

**Comment by EDLLT** [comment]
> Are you using the freeze/freeze path feature to prevent reprocessing files that have already been processed? Note that if you don't use these features, the database will be increased each time you run the flow.
Yes, I have tried the freeze feature

> Are you on the main branch locally? Please note that we haven't released the fix for this yet.
Yes, I am building langflow from the source code's main branch, commit bffb0f129bc61bacc57ec2591d3e6525e3088b93

**Comment by EDLLT** [comment]
I'll try to get to the bottom of this. Meanwhile, could this issue be reopened?

Here's a useful patch I've written for debugging purposes. It helps show how many vertex builds there are, when they get returned and when they are getting committed to the database

Here's a patch to crud.py to make it more verbose for debugging.

[verbose-vertex-crud-py.patch](https://github.com/user-attachments/files/17295417/verbose-vertex-crud-py.txt)

<details>
<summary>Example of what it outputs when building components/refreshing the page </summary>

When building a component, the data gets committed into the db
```  
/usr/lib/python3.12/asyncio/base_events.py:726: ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=False>
  _warn(f"unclosed event loop {self!r}", ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
log_vertex_build called with vertex_build id: File-osnlJ
Vertex build data size: 1816 bytes
Created VertexBuildTable with id: File-osnlJ
Table contents:
  timestamp: 2024-10-08T15:05:12.996697+00:00
  id: File-osnlJ
  data: {"results": {}, "outputs": {"data": {"message": {"file_path": "/home/edllt/.cache/langflow/b2e77365-53e4-4f56-80a9-7d51a553913d/2024-10-08_18-01-39_random_text.txt"...
  artifacts: {"data": {"rep...
  params: None
  valid: True
  flow_id: b2e77365-53e4-4f56-80a9-7d51a553913d
  build_id: dc9903b9-30d9-4fb5-97fa-b681287b097f
Successfully committed VertexBuildTable with id: File-osnlJ
log_vertex_build finished

```  

Upon refreshing, it seems like all vertex builds are being returned

```
get_vertex_builds_by_flow_id called with flow_id: b2e77365-53e4-4f56-80a9-7d51a553913d, limit: 1000
Returning 3 vertex builds
```  

</details>

Using this script, [as well as using sql commands to view the db](https://github.com/langflow-ai/langflow/issues/3970#issuecomment-2397368158), I figured that the page crash is probably because it's returning a huge amount of data upon refreshing. That along with the fact that the previously built components' cached outputs in the db don't get cleared up makes it take up a lot of storage and ram

**Comment by Cristhianzl** [comment]
@EDLLT

We have verified that there is an issue with the freeze/freeze path feature, and we are already working on it. 
I will also check into the frontend crashing.

I'll reopen the issue.
Thanks again!

**Comment by Cristhianzl** [comment]
hi @codenprogressive @EDLLT 

We have confirmed that the freeze feature is functioning as expected. I will implement a fix to optimize data storage in the database, which will reduce memory usage and prevent potential frontend crashes.

Thanks for your feedback and patience.
Feel free to contact us anytime :)

#4078

**Comment by EDLLT** [comment]
> ### Issues that still occur
> - Deleting the entire flow clears up the relevant parts of the vertex_build cache; however, deleting the file component does not delete its relevant data from the vertex_build cache. This causes ``langflow.db`` to pile up space from non-existing components overtime within the flow
> 
> - When rebuilding the file component explicitly(doesn't matter if it's frozen or not), the previously cached data does not get removed from ``vertex_build`` in ``langflow.db`` and the data gets duplicated within vertex_build.
> 
> - Uploading the same file multiple times using langflow's file component increases memory usage. In my case, uploading the ``random_ascii_70MB.txt`` 10 times increased memory usage by \~1GB
> 
> 
> ### New issues
> - When processing a large amount of data using the file component
>     -  After it has built successfully, we are able to press on Data within langflow to view the contents of our file.
>     - Then, when I try to refresh the page and try to view the data within my already built file component, it crashes my tab(tested on both brave and chrome) 
>     - ![image](https://github.com/user-attachments/assets/8260edd1-e98a-40c7-a3f1-b90e0b04df0e)
>     - ![image](https://github.com/user-attachments/assets/d605c586-39cd-411b-8ac9-f0c67210c5d9)
> 
> - Taking data from the file component then using split text processes successfully but
>     - Upon reloading the page(on the flow not the main menu), langflow starts taking up a significant amount of ram(reached up to \~15GB) before it settles down and afterwards, the browser's memory starts spiking up to 2\~3GB before crashing the tab(tested on both brave and chrome)

@Cristhianzl I haven't tested the fix yet but looking at the PR, it seems that it only addressed the frontend crashing issue through truncating long strings.

Other issues like data duplication in the database from rebuilds, dead data belonging to non-existing components when deleting them, uploading file taking memory upon each reupload, etc(i have written all the issues in the previous comment) don't seem to have been addressed

**Comment by Cristhianzl** [comment]
@EDLLT, hi

We are concerned about this and are close to the point where we will no longer use the vertex_build table. I hope that within the next few weeks, we can disable this table and eliminate data duplication.

In any case, truncating the data stored in the database (PR #4078) will drastically reduce memory usage.

**Comment by Cristhianzl** [comment]
@EDLLT hi,

I created a small flow to test the scenario you mentioned. I uploaded a 60MB CSV, which was processed and displayed correctly on my UI without any issues, even after running the flow multiple times.

There’s no problem with the freeze feature. It operates using a cache table in memory, not the vertex_build table.

My suggestion would be to clean up the vertex_build and transactions tables. Once you do that, the data will be truncated and displayed accordingly on the frontend, which should prevent any memory leaks or frontend crashes.

![image](https://github.com/user-attachments/assets/f7370b93-1bfa-48dd-a3c7-cd6d4893e0b6)

![image](https://github.com/user-attachments/assets/049e6016-1b8e-44c5-bb6d-9184ed351068)

<img width="1337" alt="image" src="https://github.com/user-attachments/assets/b7905579-dfa2-4078-9b21-a2bf7da62f5d">

<img width="1329" alt="image" src="https://github.com/user-attachments/assets/a6fb8c36-a965-4e0e-ba88-a005da87a218">

**Comment by EDLLT** [comment]
I've edited this comment to include steps to reproduce the crash
https://github.com/langflow-ai/langflow/issues/4008#issuecomment-2407550796

**Comment by Cristhianzl** [comment]
hey @EDLLT you are correct.
I found the problem. Somehow after the first run, the `params` column is been stored with memory heap, causing the cascade error to the frontend.

There’s no problem with the freeze feature. It operates using a cache table in memory, not the `vertex_build` table.

My suggestion would be to clean up the `vertex_build` and `transactions` tables. Once you do that, the data will be truncated and displayed accordingly on the frontend, which should prevent any memory leaks or frontend crashes.

We are still working on the improvement to remove this `vertex_build` table, this could take a while but, in the future we are not going to have this table anymore.

After lots of runs of this flow (this is a 50MB CSV file uploaded), the memory heap is not happening anymore :)
This field was not been validated as the others.

![image](https://github.com/user-attachments/assets/5ce13d15-01af-4117-bc2b-ea00b0e0e42f)

<img width="1323" alt="image" src="https://github.com/user-attachments/assets/80c2b519-37e0-4f62-8e39-1e1949d3a316">

I would like to thank you for your patience and help!

**Comment by Cristhianzl** [comment]
I will let this issue open until you confirm everything is working for you.

**Comment by EDLLT** [comment]
@Cristhianzl Unfortunately langflow still crashes on the latest commit d0fdc568902990142762a4ca2de3c50ca6976c28
(Btw, the first thing I had done was to clear up my transaction and vertex_build tables, create a new flow, and make sure that I am on the latest commit d0fdc568902990142762a4ca2de3c50ca6976c28)
```
sqlite> delete from "transaction";
sqlite> delete from vertex_build;
```

The steps to reproduce it remain the same
https://github.com/langflow-ai/langflow/issues/4008#issuecomment-2407550796

Also here's the random ascii file and flow(hopefully would help with reproducability)

The flow
[Langflow Crasher.json](https://github.com/user-attachments/files/17346792/Langflow.Crasher.json)

The ASCII 70MB file
https://drive.google.com/file/d/1WBp0LoZiPqCBc4IhdabtV___7ZqPaV0a/view?usp=sharing

**Comment by Cristhianzl** [comment]
@EDLLT how are you running the Langflow?

**Comment by EDLLT** [comment]
> @EDLLT how are you running the Langflow?

```
make backend on one terminal tab
make frontend on another terminal tab
```

**Comment by Cristhianzl** [comment]
hi @EDLLT,

You are absolutely correct about the error you're reporting. I'm following the steps you provided to reproduce it.

I have a solution that would temporarily fix this error. However, the real solution will be to remove the vertex_build table. So, we are going to wait until we reach the point where we can remove this table to fully resolve the problem.

For now, I ask for your patience.
Thank you!

**Comment by severfire** [comment]
Seems like i do have similar issue, also my ram usage is growing as sqlite file - now its 7GB... version of my langflow is 1.0.18 and its ran by python3.10... 

and when memory ran out:
`--- End of logging error ---
                    ERROR    2024-11-13 18:09:20 - ERROR    - utils utils.py:159
                             - Error logging transaction:
                             (sqlite3.OperationalError) disk I/O
                             error
                             (Background on this error at:
                             https://sqlalche.me/e/20/e3q8)
--- Logging error in Loguru Handler #4 ---
Record was: {'elapsed': datetime.timedelta(seconds=23076, microseconds=901699), 'exception': None, 'extra': {}, 'file': (name='utils.py', path='/home/langflow/.local/lib/python3.10/site-packages/langflow/graph/utils.py'), 'function': 'log_transaction', 'level': (name='ERROR', no=40, icon='❌'), 'line': 159, 'message': 'Error logging transaction: (sqlite3.OperationalError) disk I/O error\n(Background on this error at: https://sqlalche.me/e/20/e3q8)', 'module': 'utils', 'name': 'langflow.graph.utils', 'process': (id=3284, name='MainProcess'), 'thread': (id=139889603801088, name='MainThread'), 'time': datetime(2024, 11, 13, 18, 9, 20, 204206, tzinfo=datetime.timezone(datetime.timedelta(seconds=3600), 'CET'))}
Traceback (most recent call last):                                                                                                                                         File "/home/langflow/.local/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1144, in _commit_impl                                                            self.engine.dialect.do_commit(self.connection)                                                                                                                         File "/home/langflow/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 702, in do_commit
    dbapi_connection.commit()
sqlite3.OperationalError: disk I/O error`

**Comment by EDLLT** [comment]
> I have a solution that would temporarily fix this error. However, the real solution will be to remove the `vertex_build` table. So, we are going to wait until we reach the point where we can remove this table to fully resolve the problem.

If removing vertex build is not possible or will take significant effort, another solution I could think of would be to add checks to prevent duplicates of the same data as well as remove old cached values when adding a new value. I think this solution is simpler; however, my understanding of vertex_build is not full yet. So, did I misunderstand or will this approach work? 
Also, if I submit a PR doing this, will it be merged?

**Comment by EDLLT** [comment]
@Cristhianzl I've noticed that this issue's been closed. Has it been fixed?
Which PR fixes the issue if so?

**Comment by Cristhianzl** [comment]
I'll keep It open.
Just closed due lack of new messages.

Thanks!

**Comment by EDLLT** [comment]
@ogabrielluiz I tested this PR, the split text component fails to build now

![Image](https://github.com/user-attachments/assets/8ff7a295-f99c-4f4f-922b-66e50140179b)

Reproduction steps found here: https://github.com/langflow-ai/langflow/issues/4008#issuecomment-2407550796

```
                ╭──────────────────────────── Traceback (most recent call last) ────────────────────────────╮             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/base.py:1967 in _exec_single_context                                            │             
                             │                                                                                           │             
                             │   1964 │   │   │   │   │   │   │   evt_handled = True                                     │             
                             │   1965 │   │   │   │   │   │   │   break                                                  │             
                             │   1966 │   │   │   │   if not evt_handled:                                                │             
                             │ ❱ 1967 │   │   │   │   │   self.dialect.do_execute(                                       │             
                             │   1968 │   │   │   │   │   │   cursor, str_statement, effective_parameters, context       │             
                             │   1969 │   │   │   │   │   )                                                              │             
                             │   1970                                                                                    │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/default.py:941 in do_execute                                                    │             
                             │                                                                                           │             
                             │    938 │   │   cursor.executemany(statement, parameters)                                  │             
                             │    939 │                                                                                  │             
                             │    940 │   def do_execute(self, cursor, statement, parameters, context=None):             │             
                             │ ❱  941 │   │   cursor.execute(statement, parameters)                                      │             
                             │    942 │                                                                                  │             
                             │    943 │   def do_execute_no_params(self, cursor, statement, context=None):               │             
                             │    944 │   │   cursor.execute(statement)                                                  │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/dialects/sqlite/aiosqlite.py:147 in execute                                            │             
                             │                                                                                           │             
                             │   144 │   │   │   else:                                                                   │             
                             │   145 │   │   │   │   self._cursor = _cursor                                              │             
                             │   146 │   │   except Exception as error:                                                  │             
                             │ ❱ 147 │   │   │   self._adapt_connection._handle_exception(error)                         │             
                             │   148 │                                                                                   │             
                             │   149 │   def executemany(self, operation, seq_of_parameters):                            │             
                             │   150 │   │   try:                                                                        │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/dialects/sqlite/aiosqlite.py:298 in _handle_exception                                  │             
                             │                                                                                           │             
                             │   295 │   │   │   │   "no active connection"                                              │             
                             │   296 │   │   │   ) from error                                                            │             
                             │   297 │   │   else:                                                                       │             
                             │ ❱ 298 │   │   │   raise error                                                             │             
                             │   299                                                                                     │             
                             │   300                                                                                     │             
                             │   301 class AsyncAdaptFallback_aiosqlite_connection(AsyncAdapt_aiosqlite_connection):     │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/dialects/sqlite/aiosqlite.py:129 in execute                                            │             
                             │                                                                                           │             
                             │   126 │   │   │   if parameters is None:                                                  │             
                             │   127 │   │   │   │   self.await_(_cursor.execute(operation))                             │             
                             │   128 │   │   │   else:                                                                   │             
                             │ ❱ 129 │   │   │   │   self.await_(_cursor.execute(operation, parameters))                 │             
                             │   130 │   │   │                                                                           │             
                             │   131 │   │   │   if _cursor.description:                                                 │             
                             │   132 │   │   │   │   self.description = _cursor.description                              │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/util/_concurrency_py3k.py:132 in await_only                                            │             
                             │                                                                                           │             
                             │   129 │   # a coroutine to run. Once the awaitable is done, the driver greenlet           │             
                             │   130 │   # switches back to this greenlet with the result of awaitable that is           │             
                             │   131 │   # then returned to the caller (or raised as error)                              │             
                             │ ❱ 132 │   return current.parent.switch(awaitable)  # type: ignore[no-any-return,attr-defi │             
                             │       # noqa: E501                                                                        │             
                             │   133                                                                                     │             
                             │   134                                                                                     │             
                             │   135 def await_fallback(awaitable: Awaitable[_T]) -> _T:                                 │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/util/_concurrency_py3k.py:196 in greenlet_spawn                                        │             
                             │                                                                                           │             
                             │   193 │   │   try:                                                                        │             
                             │   194 │   │   │   # wait for a coroutine from await_only and then return its              │             
                             │   195 │   │   │   # result back to it.                                                    │             
                             │ ❱ 196 │   │   │   value = await result                                                    │             
                             │   197 │   │   except BaseException:                                                       │             
                             │   198 │   │   │   # this allows an exception to be raised within                          │             
                             │   199 │   │   │   # the moderated greenlet so that it can continue                        │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/aiosqlit │             
                             │ e/cursor.py:48 in execute                                                                 │             
                             │                                                                                           │             
                             │    45 │   │   """Execute the given query."""                                              │             
                             │    46 │   │   if parameters is None:                                                      │             
                             │    47 │   │   │   parameters = []                                                         │             
                             │ ❱  48 │   │   await self._execute(self._cursor.execute, sql, parameters)                  │             
                             │    49 │   │   return self                                                                 │             
                             │    50 │                                                                                   │             
                             │    51 │   async def executemany(                                                          │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/aiosqlit │             
                             │ e/cursor.py:40 in _execute                                                                │             
                             │                                                                                           │             
                             │    37 │                                                                                   │             
                             │    38 │   async def _execute(self, fn, *args, **kwargs):                                  │             
                             │    39 │   │   """Execute the given function on the shared connection's thread."""         │             
                             │ ❱  40 │   │   return await self._conn._execute(fn, *args, **kwargs)                       │             
                             │    41 │                                                                                   │             
                             │    42 │   async def execute(                                                              │             
                             │    43 │   │   self, sql: str, parameters: Optional[Iterable[Any]] = None                  │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/aiosqlit │             
                             │ e/core.py:132 in _execute                                                                 │             
                             │                                                                                           │             
                             │   129 │   │                                                                               │             
                             │   130 │   │   self._tx.put_nowait((future, function))                                     │             
                             │   131 │   │                                                                               │             
                             │ ❱ 132 │   │   return await future                                                         │             
                             │   133 │                                                                                   │             
                             │   134 │   async def _connect(self) -> "Connection":                                       │             
                             │   135 │   │   """Connect to the actual sqlite database."""                                │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/aiosqlit │             
                             │ e/core.py:115 in run                                                                      │             
                             │                                                                                           │             
                             │   112 │   │   │                                                                           │             
                             │   113 │   │   │   try:                                                                    │             
                             │   114 │   │   │   │   LOG.debug("executing %s", function)                                 │             
                             │ ❱ 115 │   │   │   │   result = function()                                                 │             
                             │   116 │   │   │   │   LOG.debug("operation %s completed", function)                       │             
                             │   117 │   │   │   │   future.get_loop().call_soon_threadsafe(set_result, future, result)  │             
                             │   118 │   │   │   except BaseException as e:  # noqa B036                                 │             
                             ╰───────────────────────────────────────────────────────────────────────────────────────────╯             
                             DataError: string or blob too big                                                                         
                                                                                                                                       
                             The above exception was the direct cause of the following exception:                                      
                                                                                                                                       
                             ╭──────────────────────────── Traceback (most recent call last) ────────────────────────────╮             
                             │ /home/edllt/src/langflow/langflow_source-code/src/backend/base/langflow/graph/utils.py:19 │             
                             │ 1 in log_vertex_build                                                                     │             
                             │                                                                                           │             
                             │   188 │   │   │   artifacts=json.loads(json.dumps(artifacts, default=str)),               │             
                             │   189 │   │   )                                                                           │             
                             │   190 │   │   async with session_getter(get_db_service()) as session:                     │             
                             │ ❱ 191 │   │   │   inserted = await crud_log_vertex_build(session, vertex_build)           │             
                             │   192 │   │   │   logger.debug(f"Logged vertex build: {inserted.build_id}")               │             
                             │   193 │   except Exception:  # noqa: BLE001                                               │             
                             │   194 │   │   logger.exception("Error logging vertex build")                              │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/src/backend/base/langflow/services/database │             
                             │ /models/vertex_builds/crud.py:95 in log_vertex_build                                      │             
                             │                                                                                           │             
                             │    92 │   │                                                                               │             
                             │    93 │   │   # 1) Insert and flush the new build so queries can see it                   │             
                             │    94 │   │   db.add(table)                                                               │             
                             │ ❱  95 │   │   await db.flush()                                                            │             
                             │    96 │   │                                                                               │             
                             │    97 │   │   # 2) Delete older builds for this vertex, keeping newest max_per_vertex     │             
                             │    98 │   │   keep_vertex_subq = (                                                        │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/ext/asyncio/session.py:800 in flush                                                    │             
                             │                                                                                           │             
                             │    797 │   │   │   :meth:`_orm.Session.flush` - main documentation for flush              │             
                             │    798 │   │                                                                              │             
                             │    799 │   │   """                                                                        │             
                             │ ❱  800 │   │   await greenlet_spawn(self.sync_session.flush, objects=objects)             │             
                             │    801 │                                                                                  │             
                             │    802 │   def get_transaction(self) -> Optional[AsyncSessionTransaction]:                │             
                             │    803 │   │   """Return the current root transaction in progress, if any.                │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/util/_concurrency_py3k.py:203 in greenlet_spawn                                        │             
                             │                                                                                           │             
                             │   200 │   │   │   # its expected flow.                                                    │             
                             │   201 │   │   │   result = context.throw(*sys.exc_info())                                 │             
                             │   202 │   │   else:                                                                       │             
                             │ ❱ 203 │   │   │   result = context.switch(value)                                          │             
                             │   204 │                                                                                   │             
                             │   205 │   if _require_await and not switch_occurred:                                      │             
                             │   206 │   │   raise exc.AwaitRequired(                                                    │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/orm/session.py:4352 in flush                                                           │             
                             │                                                                                           │             
                             │   4349 │   │   │   return                                                                 │             
                             │   4350 │   │   try:                                                                       │             
                             │   4351 │   │   │   self._flushing = True                                                  │             
                             │ ❱ 4352 │   │   │   self._flush(objects)                                                   │             
                             │   4353 │   │   finally:                                                                   │             
                             │   4354 │   │   │   self._flushing = False                                                 │             
                             │   4355                                                                                    │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/orm/session.py:4487 in _flush                                                          │             
                             │                                                                                           │             
                             │   4484 │   │   │   transaction.commit()                                                   │             
                             │   4485 │   │                                                                              │             
                             │   4486 │   │   except:                                                                    │             
                             │ ❱ 4487 │   │   │   with util.safe_reraise():                                              │             
                             │   4488 │   │   │   │   transaction.rollback(_capture_exception=True)                      │             
                             │   4489 │                                                                                  │             
                             │   4490 │   def bulk_save_objects(                                                         │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/util/langhelpers.py:146 in __exit__                                                    │             
                             │                                                                                           │             
                             │    143 │   │   │   exc_type, exc_value, exc_tb = self._exc_info                           │             
                             │    144 │   │   │   assert exc_value is not None                                           │             
                             │    145 │   │   │   self._exc_info = None  # remove potential circular references          │             
                             │ ❱  146 │   │   │   raise exc_value.with_traceback(exc_tb)                                 │             
                             │    147 │   │   else:                                                                      │             
                             │    148 │   │   │   self._exc_info = None  # remove potential circular references          │             
                             │    149 │   │   │   assert value is not None                                               │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/orm/session.py:4448 in _flush                                                          │             
                             │                                                                                           │             
                             │   4445 │   │   try:                                                                       │             
                             │   4446 │   │   │   self._warn_on_events = True                                            │             
                             │   4447 │   │   │   try:                                                                   │             
                             │ ❱ 4448 │   │   │   │   flush_context.execute()                                            │             
                             │   4449 │   │   │   finally:                                                               │             
                             │   4450 │   │   │   │   self._warn_on_events = False                                       │             
                             │   4451                                                                                    │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/orm/unitofwork.py:466 in execute                                                       │             
                             │                                                                                           │             
                             │   463 │   │   │   │   │   n.execute_aggregate(self, set_)                                 │             
                             │   464 │   │   else:                                                                       │             
                             │   465 │   │   │   for rec in topological.sort(self.dependencies, postsort_actions):       │             
                             │ ❱ 466 │   │   │   │   rec.execute(self)                                                   │             
                             │   467 │                                                                                   │             
                             │   468 │   def finalize_flush_changes(self) -> None:                                       │             
                             │   469 │   │   """Mark processed objects as clean / deleted after a successful             │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/orm/unitofwork.py:642 in execute                                                       │             
                             │                                                                                           │             
                             │   639 │                                                                                   │             
                             │   640 │   @util.preload_module("sqlalchemy.orm.persistence")                              │             
                             │   641 │   def execute(self, uow):                                                         │             
                             │ ❱ 642 │   │   util.preloaded.orm_persistence.save_obj(                                    │             
                             │   643 │   │   │   self.mapper,                                                            │             
                             │   644 │   │   │   uow.states_for_mapper_hierarchy(self.mapper, False, False),             │             
                             │   645 │   │   │   uow,                                                                    │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/orm/persistence.py:93 in save_obj                                                      │             
                             │                                                                                           │             
                             │     90 │   │   │   update,                                                                │             
                             │     91 │   │   )                                                                          │             
                             │     92 │   │                                                                              │             
                             │ ❱   93 │   │   _emit_insert_statements(                                                   │             
                             │     94 │   │   │   base_mapper,                                                           │             
                             │     95 │   │   │   uowtransaction,                                                        │             
                             │     96 │   │   │   mapper,                                                                │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/orm/persistence.py:1048 in _emit_insert_statements                                     │             
                             │                                                                                           │             
                             │   1045 │   │   │   records = list(records)                                                │             
                             │   1046 │   │   │   multiparams = [rec[2] for rec in records]                              │             
                             │   1047 │   │   │                                                                          │             
                             │ ❱ 1048 │   │   │   result = connection.execute(                                           │             
                             │   1049 │   │   │   │   statement, multiparams, execution_options=execution_options        │             
                             │   1050 │   │   │   )                                                                      │             
                             │   1051 │   │   │   if bookkeeping:                                                        │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/base.py:1418 in execute                                                         │             
                             │                                                                                           │             
                             │   1415 │   │   except AttributeError as err:                                              │             
                             │   1416 │   │   │   raise exc.ObjectNotExecutableError(statement) from err                 │             
                             │   1417 │   │   else:                                                                      │             
                             │ ❱ 1418 │   │   │   return meth(                                                           │             
                             │   1419 │   │   │   │   self,                                                              │             
                             │   1420 │   │   │   │   distilled_parameters,                                              │             
                             │   1421 │   │   │   │   execution_options or NO_OPTIONS,                                   │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/sql/elements.py:515 in _execute_on_connection                                          │             
                             │                                                                                           │             
                             │    512 │   │   if self.supports_execution:                                                │             
                             │    513 │   │   │   if TYPE_CHECKING:                                                      │             
                             │    514 │   │   │   │   assert isinstance(self, Executable)                                │             
                             │ ❱  515 │   │   │   return connection._execute_clauseelement(                              │             
                             │    516 │   │   │   │   self, distilled_params, execution_options                          │             
                             │    517 │   │   │   )                                                                      │             
                             │    518 │   │   else:                                                                      │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/base.py:1640 in _execute_clauseelement                                          │             
                             │                                                                                           │             
                             │   1637 │   │   │   schema_translate_map=schema_translate_map,                             │             
                             │   1638 │   │   │   linting=self.dialect.compiler_linting | compiler.WARN_LINTING,         │             
                             │   1639 │   │   )                                                                          │             
                             │ ❱ 1640 │   │   ret = self._execute_context(                                               │             
                             │   1641 │   │   │   dialect,                                                               │             
                             │   1642 │   │   │   dialect.execution_ctx_cls._init_compiled,                              │             
                             │   1643 │   │   │   compiled_sql,                                                          │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/base.py:1846 in _execute_context                                                │             
                             │                                                                                           │             
                             │   1843 │   │   if context.execute_style is ExecuteStyle.INSERTMANYVALUES:                 │             
                             │   1844 │   │   │   return self._exec_insertmany_context(dialect, context)                 │             
                             │   1845 │   │   else:                                                                      │             
                             │ ❱ 1846 │   │   │   return self._exec_single_context(                                      │             
                             │   1847 │   │   │   │   dialect, context, statement, parameters                            │             
                             │   1848 │   │   │   )                                                                      │             
                             │   1849                                                                                    │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/base.py:1986 in _exec_single_context                                            │             
                             │                                                                                           │             
                             │   1983 │   │   │   result = context._setup_result_proxy()                                 │             
                             │   1984 │   │                                                                              │             
                             │   1985 │   │   except BaseException as e:                                                 │             
                             │ ❱ 1986 │   │   │   self._handle_dbapi_exception(                                          │             
                             │   1987 │   │   │   │   e, str_statement, effective_parameters, cursor, context            │             
                             │   1988 │   │   │   )                                                                      │             
                             │   1989                                                                                    │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/base.py:2355 in _handle_dbapi_exception                                         │             
                             │                                                                                           │             
                             │   2352 │   │   │   │   raise newraise.with_traceback(exc_info[2]) from e                  │             
                             │   2353 │   │   │   elif should_wrap:                                                      │             
                             │   2354 │   │   │   │   assert sqlalchemy_exception is not None                            │             
                             │ ❱ 2355 │   │   │   │   raise sqlalchemy_exception.with_traceback(exc_info[2]) from e      │             
                             │   2356 │   │   │   else:                                                                  │             
                             │   2357 │   │   │   │   assert exc_info[1] is not None                                     │             
                             │   2358 │   │   │   │   raise exc_info[1].with_traceback(exc_info[2])                      │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/base.py:1967 in _exec_single_context                                            │             
                             │                                                                                           │             
                             │   1964 │   │   │   │   │   │   │   evt_handled = True                                     │             
                             │   1965 │   │   │   │   │   │   │   break                                                  │             
                             │   1966 │   │   │   │   if not evt_handled:                                                │             
                             │ ❱ 1967 │   │   │   │   │   self.dialect.do_execute(                                       │             
                             │   1968 │   │   │   │   │   │   cursor, str_statement, effective_parameters, context       │             
                             │   1969 │   │   │   │   │   )                                                              │             
                             │   1970                                                                                    │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/engine/default.py:941 in do_execute                                                    │             
                             │                                                                                           │             
                             │    938 │   │   cursor.executemany(statement, parameters)                                  │             
                             │    939 │                                                                                  │             
                             │    940 │   def do_execute(self, cursor, statement, parameters, context=None):             │             
                             │ ❱  941 │   │   cursor.execute(statement, parameters)                                      │             
                             │    942 │                                                                                  │             
                             │    943 │   def do_execute_no_params(self, cursor, statement, context=None):               │             
                             │    944 │   │   cursor.execute(statement)                                                  │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/dialects/sqlite/aiosqlite.py:147 in execute                                            │             
                             │                                                                                           │             
                             │   144 │   │   │   else:                                                                   │             
                             │   145 │   │   │   │   self._cursor = _cursor                                              │             
                             │   146 │   │   except Exception as error:                                                  │             
                             │ ❱ 147 │   │   │   self._adapt_connection._handle_exception(error)                         │             
                             │   148 │                                                                                   │             
                             │   149 │   def executemany(self, operation, seq_of_parameters):                            │             
                             │   150 │   │   try:                                                                        │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/dialects/sqlite/aiosqlite.py:298 in _handle_exception                                  │             
                             │                                                                                           │             
                             │   295 │   │   │   │   "no active connection"                                              │             
                             │   296 │   │   │   ) from error                                                            │             
                             │   297 │   │   else:                                                                       │             
                             │ ❱ 298 │   │   │   raise error                                                             │             
                             │   299                                                                                     │             
                             │   300                                                                                     │             
                             │   301 class AsyncAdaptFallback_aiosqlite_connection(AsyncAdapt_aiosqlite_connection):     │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/dialects/sqlite/aiosqlite.py:129 in execute                                            │             
                             │                                                                                           │             
                             │   126 │   │   │   if parameters is None:                                                  │             
                             │   127 │   │   │   │   self.await_(_cursor.execute(operation))                             │             
                             │   128 │   │   │   else:                                                                   │             
                             │ ❱ 129 │   │   │   │   self.await_(_cursor.execute(operation, parameters))                 │             
                             │   130 │   │   │                                                                           │             
                             │   131 │   │   │   if _cursor.description:                                                 │             
                             │   132 │   │   │   │   self.description = _cursor.description                              │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/util/_concurrency_py3k.py:132 in await_only                                            │             
                             │                                                                                           │             
                             │   129 │   # a coroutine to run. Once the awaitable is done, the driver greenlet           │             
                             │   130 │   # switches back to this greenlet with the result of awaitable that is           │             
                             │   131 │   # then returned to the caller (or raised as error)                              │             
                             │ ❱ 132 │   return current.parent.switch(awaitable)  # type: ignore[no-any-return,attr-defi │             
                             │       # noqa: E501                                                                        │             
                             │   133                                                                                     │             
                             │   134                                                                                     │             
                             │   135 def await_fallback(awaitable: Awaitable[_T]) -> _T:                                 │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/sqlalche │             
                             │ my/util/_concurrency_py3k.py:196 in greenlet_spawn                                        │             
                             │                                                                                           │             
                             │   193 │   │   try:                                                                        │             
                             │   194 │   │   │   # wait for a coroutine from await_only and then return its              │             
                             │   195 │   │   │   # result back to it.                                                    │             
                             │ ❱ 196 │   │   │   value = await result                                                    │             
                             │   197 │   │   except BaseException:                                                       │             
                             │   198 │   │   │   # this allows an exception to be raised within                          │             
                             │   199 │   │   │   # the moderated greenlet so that it can continue                        │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/aiosqlit │             
                             │ e/cursor.py:48 in execute                                                                 │             
                             │                                                                                           │             
                             │    45 │   │   """Execute the given query."""                                              │             
                             │    46 │   │   if parameters is None:                                                      │             
                             │    47 │   │   │   parameters = []                                                         │             
                             │ ❱  48 │   │   await self._execute(self._cursor.execute, sql, parameters)                  │             
                             │    49 │   │   return self                                                                 │             
                             │    50 │                                                                                   │             
                             │    51 │   async def executemany(                                                          │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/aiosqlit │             
                             │ e/cursor.py:40 in _execute                                                                │             
                             │                                                                                           │             
                             │    37 │                                                                                   │             
                             │    38 │   async def _execute(self, fn, *args, **kwargs):                                  │             
                             │    39 │   │   """Execute the given function on the shared connection's thread."""         │             
                             │ ❱  40 │   │   return await self._conn._execute(fn, *args, **kwargs)                       │             
                             │    41 │                                                                                   │             
                             │    42 │   async def execute(                                                              │             
                             │    43 │   │   self, sql: str, parameters: Optional[Iterable[Any]] = None                  │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/aiosqlit │             
                             │ e/core.py:132 in _execute                                                                 │             
                             │                                                                                           │             
                             │   129 │   │                                                                               │             
                             │   130 │   │   self._tx.put_nowait((future, function))                                     │             
                             │   131 │   │                                                                               │             
                             │ ❱ 132 │   │   return await future                                                         │             
                             │   133 │                                                                                   │             
                             │   134 │   async def _connect(self) -> "Connection":                                       │             
                             │   135 │   │   """Connect to the actual sqlite database."""                                │             
                             │                                                                                           │             
                             │ /home/edllt/src/langflow/langflow_source-code/.venv/lib/python3.12/site-packages/aiosqlit │             
                             │ e/core.py:115 in run                                                                      │             
                             │                                                                                           │             
                             │   112 │   │   │                                                                           │             
                             │   113 │   │   │   try:                                                                    │             
                             │   114 │   │   │   │   LOG.debug("executing %s", function)                                 │             
                             │ ❱ 115 │   │   │   │   result = function()                                                 │             
                             │   116 │   │   │   │   LOG.debug("operation %s completed", function)                       │             
                             │   117 │   │   │   │   future.get_loop().call_soon_threadsafe(set_result, future, result)  │             
                             │   118 │   │   │   except BaseException as e:  # noqa B036                                 │             
                             ╰───────────────────────────────────────────────────────────────────────────────────────────╯             
                             DataError: (sqlite3.DataError) string or blob too big                                                     
                             [SQL: INSERT INTO vertex_build (timestamp, id, data, artifacts, params, valid, flow_id,                   
                             build_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?)]                                                                
                             [parameters: ('2025-01-20 14:54:23.916130', 'SplitText-DbSks', '{"results": {}, "outputs":                
                             {"chunks": {"message": [{"text_key": "text", "data": {"file_path":                                        
                             "/home/edllt/.cache/langflow/fafa1015-141f-4dac-9e70-03 ... (1043500995 characters truncated)             
                             ... \\\\{wtkmXF|Z"}, "default_value": ""}], "type": "array"}}, "timedelta":                               
                             15.242670240986627, "duration": "15.24 seconds", "used_frozen_result": false}', '{"chunks":               
                             {"repr": ["{\\n    \\"file_path\\":                                                                       
                             \\"/home/edllt/.cache/langflow/fafa1015-141f-4dac-9e70-0388e3bb1841/2025-01-20_17-51-20_rando             
                             m_ascii_w ... (431031245 characters truncated) ...                                                        
                             #,=PJ7KHG\\"\'SRPs03[v5ys*#r>R*6Q\\"k03SNS0U4MjjDn\\"Exya(d19\'_:]Uza1557u:;q3hwDIdSpR1LBY<:\             
                             \\\{wtkmXF|Z"}, "default_value": ""}], "type": "array"}}', 'None', 1,                                     
                             'fafa1015141f4dac9e700388e3bb1841', 'd753b88412404459ab57bb00d172f4b4')]                                  
                             (Background on this error at: https://sqlalche.me/e/20/9h9h)
```

**Comment by severfire** [comment]
@Cristhianzl worth checking

**Comment by EDLLT** [comment]
My issue has been closed despite not being fixed and I'm not able to reopen it as I don't have permissions

**Comment by ogabrielluiz** [comment]
That's really weird. I'll take another look.

**Comment by EDLLT** [comment]
> [@EDLLT](https://github.com/EDLLT) Is it the same issue ? It seems different: memory ballooning vs `DataError: (sqlite3.DataError) string or blob too big`

Hmm, after looking more closely at the error, that's true. But, at the same time, since the component fails to build, I am unable to check and test whether if the memory ballooning doesn't actually occur anymore
